src/Pos/Wallet/Web/ClientTypes.hs

```diff
@@ -101,28 +111,38 @@ data CCurrency
     deriving (Show, Read, Generic)
 
 -- | Client hash
-newtype CHash = CHash Text deriving (Show, Eq, Generic, Buildable)
+newtype CHash = CHash Text
+    deriving (Show, Eq, Generic, Buildable)
 
 instance Hashable CHash where
     hashWithSalt s (CHash h) = hashWithSalt s h
 
 -- | Client address
-newtype CAddress = CAddress CHash deriving (Show, Eq, Generic, Hashable, Buildable)
+-- @w@ is phantom type and stands for type of item this address belongs to.
+newtype CAddress w = CAddress CHash
```

Comment:
Ahah, this makes sense for me. We can make type aliases for different types of `CAddress` then (`type CAccAddress = CAddress 'Acc` ?), because I'm lazy to type apostrophe each time.
I saw @akegalj is performing refactoring in another branch, if I introduce this changes now, his merge would turn into entire conflict. Let's add it after @akegalj merges that refactoring back